### ISSUE TITLE
Consider removing `RawContextMiddleware` and keeping only `ContextMiddleware` after resolving memory usage issue

### ISSUE THREAD

**Issue by tomwojcik** [tomwojcik/starlette-context#47]
I like `ContextMiddleware` because it's very simple to understand and expand, even for minds that are not familiar with `async` Python.

Some time ago it was advised https://github.com/tomwojcik/starlette-context/issues/18 to add `RawContextMiddleware` and Starlette maintainers were discouraging the use of the built-in middleware. I was surprised to see 3 ❤️ reactions under this issue so I think a few people had problems with it.

Fast-forward almost a year, the issue https://github.com/encode/starlette/issues/1012#issuecomment-866622798 with memory usage has been closed. 

If there's no point in keeping both, I'd be happy to remove the more complicated one but I will wait for some confirmations from the community. If there's a case when it makes sense to use `RawContextMiddleware`, then I'd keep both.

What's your opinion about that? @hhamana @dmig-alarstudios

**Comment by hhamana** [comment]
I haven't seen much difference with either middleware. If the original concern is now fixed, the regular `ContextMiddleware ` is indeed much easier to work with, so supporting only this one would surely simplify things.
But both are working fine so far, so I don't see a pressing reason to remove it either.
Maybe when there are some differences that can't be abstracted nicely?

**Comment by michaeltoohig** [comment]
If the original concern is fixed then the docs need to be updated. I ended up in this issue while trying to figure out what was wrong with `ContextMiddleware` from reading the warning in the docs [here](https://starlette-context.readthedocs.io/en/latest/middleware.html#why-are-there-two-middlewares-that-do-the-same-thing)

**Comment by adriangb** [comment]
Hi I am a Starlette maintainer. I _highly_ encourage you to not use use `BaseHTTPMiddleware`. I would stick with `RawContextMiddleware` and deprecate `ContextMiddleware`. cc @klundex

**Comment by tomwojcik** [comment]
Great! I was hoping one of the Starlette maintainers would chime in and help make this decision. Thanks.

**Comment by flipbit03** [comment]
@adriangb can you clarify why would it be encouraged to keep using `RawContextMiddleware` over the other? What are the other's shortcomings? Can we simply phase out the other one, if it's a case of being defective? Thanks in advance

**Comment by adriangb** [comment]
I think it's best to link to discussions already existing in Starlette: https://github.com/encode/starlette/discussions/1729. There are really no benefits to `starlette-context`'s users for having both middlewares, there is only potential downsides that they will run into bugs related to BaseHTTPMiddleware.

**Comment by tomwojcik** [comment]
@adriangb sure, that's why in the next minor release we're getting rid of it. The deprecation warning is already there.

Thanks for linking the related discussion.

**Comment by flipbit03** [comment]
Thank you @adriangb for the context link!

**Comment by Kludex** [comment]
Yes, it is. Sorry the delay, AI is giving me a lot of work lately.

Let me know if you need a review or want to talk more about it.
